dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/MetadataFormLookup.java

```diff
@@ -0,0 +1,68 @@
+package edu.harvard.iq.dataverse.importers.ui;
+
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldType;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldsByType;
+import edu.harvard.iq.dataverse.persistence.dataset.MetadataBlock;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class MetadataFormLookup {
```

Comment:
I could also use some Javadoc, the task is pretty complex so explaining context in each step is helpful.